### PR TITLE
chore(*): update deps

### PR DESCRIPTION
**Pull request by muhabdulkadir**
# Closes #<CORRESPONDING ISSUE NUMBER>

### Changes

- Update all essential deps
- Remove unused dependencies

### Author Checklist
- [ ] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

**Review by jamieshorten — APPROVED** [review]

File: packages/concerto-analysis/package.json [hunk 1]
```diff
@@ -32,24 +32,17 @@
...
-        "@types/semver": "7.3.10",
-        "@typescript-eslint/eslint-plugin": "5.27.1",
-        "@typescript-eslint/parser": "5.27.1",
-        "eslint": "8.2.0",
+        "@types/semver": "7.5.8",
+        "@typescript-eslint/eslint-plugin": "8.16.0",
+        "@typescript-eslint/parser": "8.16.0",
+        "eslint": "8.57.1",
         "jest": "^29.7.0",
-        "lodash.memoize": "4.1.2",
```
Comment (sanketshevkar) [inline packages/concerto-analysis/package.json:46]: Are these unused packages?

File: packages/concerto-cto/package.json [hunk 2]
```diff
@@ -38,24 +38,21 @@
...
-    "peggy": "2.0.1",
-    "sinon": "12.0.0",
-    "sinon-chai": "3.7.0",
-    "typescript": "5.6.3"
+    "debug": "4.3.7",
+    "eslint": "8.57.1",
+    "jsdoc": "^4.0.4",
+    "mocha": "10.8.2",
+    "nyc": "17.1.0",
+    "peggy": "4.2.0",
```
Comment (sanketshevkar) [inline packages/concerto-cto/package.json:49]: @dselman @mttrbrts do we want to update the version for peggy? It's leading to quite a lot of changes to the generated parser file. Although all the unit tests are passing.

File: packages/concerto-cto/types/lib/parser.d.ts [hunk 3]
```diff
@@ -7,4 +7,5 @@ declare namespace peg$SyntaxError {
     function buildMessage(expected: any, found: any): string;
 }
 declare function peg$parse(input: any, options: any): any;
+export declare let StartRules: string[];
```
Comment (sanketshevkar) [inline packages/concerto-cto/types/lib/parser.d.ts:10]: @muhabdulkadir is this auto generated?

File: packages/concerto-analysis/package.json (hunk 1 above)
Comment (muhabdulkadir) [reply packages/concerto-analysis/package.json:46]: Yes it is.

File: packages/concerto-cto/package.json (hunk 2 above)
Comment (mttrbrts) [reply packages/concerto-cto/package.json:49]: I've reviewed the release notes for v3.0 and v4.0 of peggy, and am comfortable that the changes don't affect us.

Providing that the tests and build pass, I'm happy to proceed.

**Pull request merged**